modules/builds-identifying-image-change-triggers.adoc

```diff
@@ -0,0 +1,79 @@
+// Module included in the following assemblies:
+//
+// * builds/triggering-builds-build-hooks.adoc
+
+[id="builds-image-change-trigger-identification_{context}"]
+= Image change trigger identification
+
+As a developer, if you have configured image change triggers, you can identify which image change initiated the last build.
+
+To accomplish this, you must identify elements in your build configuration's specification and status that are related to image change triggers.
+
+This way, you can use the timestamp in `buildConfig.status.imageChangeTriggers` to identify the most recent build. Then you can use the name and namespace of the image stream that triggered this build to find the corresponding image change trigger in the `buildConfig.spec.triggers`.
+
+
+== Image change trigger elements in the specification
+
+In your build configuration specification, `buildConfig.spec.triggers` is an array of build trigger policies, `BuildTriggerPolicy`.
+
+Each `BuildTriggerPolicy` has a `type` field and set of pointers fields, where each pointer field corresponds to one of the allowed values for the `type` field. As such, only one pointer field can be set for a given `BuildTriggerPolicy`.
+
+So for image change triggers, the value of `type` is `ImageChange`.
+
+Then, the `imageChange` field is the pointer to an `ImageChangeTrigger` object. So this will be set. It has the following fields:
+
+* `lastTriggeredImageID`: This field is deprecated in {product-title} 4.8, but is still being set. It will be ignored in a future release. It contains the resolved image reference for the `ImageStreamTag` when the last build was triggered from this `BuildConfig`.
+* `paused`: This field is used to temporarily disable this particular image change trigger.
+* `from`: This field is used to reference the `ImageStreamTag` that drives this image change trigger. Its type is the core Kubernetes type, `OwnerReference`. The `from` field has the following fields of note:
+  * `kind`: In this case, the only supported value is `ImageStreamTag`.
+  * `namespace`: The namespace where the `ImageStreamTag` lives.
+  * `name`: The name of the `ImageStreamTag`.
+
+The following example shows the relative location of the elements mentioned in the preceding list and omits unrelated elements, such as `name`, `source`, and `strategy`.
```

Comment:
The example really only includes the `from` field.  Assuming this is a YAML file, it would be helpful to include all of the described parameters from the above paragraph, if only so that users can see the proper nesting/indenation for each element if they want to include them.  (If this is a generated/read-only file, ignore this comment.)

Reply:
they would not be including pause if they are using this at all, and we don't want them setting last triggered image ID (the fact that was mistakingly put into spec in 2014/2015 is the reason we are in this mess)

we in fact in 4.7 added firing a k8s event saying don't set last triggered image id

if anything, remove the paused and last triggered image id descriptions above

but in the context of what users messing with this know from our previous user interactions on this, what we have is fine IMO and we should leave this as is